phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewTTLIT.java

```diff
@@ -0,0 +1,111 @@
+import org.apache.hadoop.hbase.HConstants;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.client.ResultScanner;
+import org.apache.hadoop.hbase.client.Scan;
+import org.apache.hadoop.hbase.client.Table;
+import org.apache.hadoop.hbase.filter.*;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.phoenix.jdbc.PhoenixDatabaseMetaData;
+import org.apache.phoenix.query.PhoenixTestBuilder;
+import org.apache.phoenix.query.QueryConstants;
+import org.apache.phoenix.util.SchemaUtil;
+import org.apache.phoenix.util.TestUtil;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class ViewTTLIT extends ParallelStatsDisabledIT {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ViewTTLIT.class);
+
+    // Scans the HBase rows directly for the view ttl related header rows column and asserts
+    private void assertViewHeaderRowsHaveViewTTLRelatedCells(String schemaName, long minTimestamp,
+            int expectedRows) throws IOException, SQLException {
+
+        FilterList filterList = new FilterList(FilterList.Operator.MUST_PASS_ALL);
+        RowFilter schemaNameFilter = new RowFilter(
+                CompareFilter.CompareOp.EQUAL,
+                new SubstringComparator(schemaName)
+        );
+        ColumnPrefixFilter viewTTLPrefixFilter = new ColumnPrefixFilter(PhoenixDatabaseMetaData.VIEW_TTL_BYTES);
+        filterList.addFilter(schemaNameFilter);
+        filterList.addFilter(viewTTLPrefixFilter);
+        try (Table tbl = driver.getConnectionQueryServices(getUrl(), TestUtil.TEST_PROPERTIES)
+                .getTable(PhoenixDatabaseMetaData.SYSTEM_CATALOG_NAME_BYTES)) {
+
+            Scan allRows = new Scan();
+            allRows.setRaw(true);
+            allRows.setTimeRange(minTimestamp, HConstants.LATEST_TIMESTAMP);
+            allRows.setFilter(filterList);
+            ResultScanner scanner = tbl.getScanner(allRows);
+            int numMatchingRows = 0;
+            for (Result result = scanner.next(); result != null; result = scanner.next()) {
+                numMatchingRows +=
+                        result.containsColumn(QueryConstants.DEFAULT_COLUMN_FAMILY_BYTES,
+                                PhoenixDatabaseMetaData.VIEW_TTL_BYTES) ? 1 : 0;
+            }
+            assertEquals(String.format("Expected rows do match for table = %s at timestamp %d",
+                    Bytes.toString(PhoenixDatabaseMetaData.SYSTEM_CATALOG_NAME_BYTES), minTimestamp), expectedRows, numMatchingRows);
+        }
+
+    }
+
+    /**
+         * -----------------
+         * Test methods
+         * -----------------
+         */
+
+    @Test
+    public void testWithBasicTenantViewWithNoViewTTLDefined() throws Exception {
```

Comment:
Do we also want to add a test for a case where we specify a viewTTL?

Reply:
I guess that will be a TODO..?

Reply:
Yeah, planning to do that as part of my next PR - which is tying in the actual property value specified during the create/alter command.